server/lib/transferwise.ts

```diff
@@ -66,15 +73,32 @@ const parseError = (
 
   return new TransferwiseError(message, code);
 };
-const requestDataAndThrowParsedError = async (request: Promise<any>, defaultErrorMessage?: string): Promise<any> => {
+export const requestDataAndThrowParsedError = async (
+  fn: Function,
+  url: string,
+  { data, ...options }: { data?: object; headers: object; params?: object },
+  defaultErrorMessage?: string,
+): Promise<any> => {
   try {
-    const response = await request;
+    const response = data ? await fn(url, data, options) : await fn(url, options);
     return getData(response);
   } catch (e) {
-    debug(e.response?.data || e);
-    const error = parseError(e, defaultErrorMessage);
-    logger.error(error.toString());
-    throw error;
+    // Implements Strong Customer Authentication
+    // https://api-docs.transferwise.com/#payouts-guide-strong-customer-authentication
+    const signatureFailed = e?.response?.headers['x-2fa-approval-result'] === 'REJECTED';
+    const hadSignature = e?.response?.headers['X-Signature'];
+    if (signatureFailed && !hadSignature) {
+      const ott = e.response.headers['x-2fa-approval'];
+      const signature = signString(ott);
+      options.headers = { ...options.headers, 'X-Signature': signature, 'x-2fa-approval': ott };
+      const response = data ? await fn(url, data, options) : await fn(url, options);
```

Comment:
If this second request fails, will it be logged properly?

Reply:
Nice catch!

Reply:
@Betree https://github.com/opencollective/opencollective-api/pull/4256